common/observers/fs/fs.go

```diff
@@ -0,0 +1,127 @@
+package fs
+
+import (
+	"bytes"
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/logging"
+	"gopkg.in/fsnotify/fsnotify.v1"
+	"io/ioutil"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// WatcherCallback described the interface for callback on matched event
+type WatcherCallback = func(details Event)
+
+// Op mimics fsnotify.Op for backward compatibility with fsnotify
+type Op fsnotify.Op
+
+// Event mimics fsnotify.Event for backward compatibility with fsnotify
+type Event fsnotify.Event
+
+// Watcher is an implementation of a wrapper around fs events
+type Watcher struct {
+	sync.Mutex
+	eventHandlers map[Op]WatcherCallback
+	fileInfo      map[string]FileInfo
+	state         int64
+}
+
+// FileInfo is a structure which describes file being tracked
+type FileInfo struct {
+	Path    string
+	Content []byte
+}
+
+// set of constants which almost mimics fsnotify type of events, except ChangedContent
+const (
+	Create Op = 1 << iota
+	Write
+	Remove
+	Rename
+	Chmod
+	ChangedContent
```

Comment:
only this one is used why ?

Reply:
basically, my problem I tried to solve is following: when operator executes in bash console ifconfig SRIOV_PHYS_DEVICE up/down then kernel event doesn't occur, so we are not able to react appropriately. First of all I tried to use fsnotifier to watch for the file: /sys/class/net/SRIOV_PHYS_DEVICE/operstate but it doesn't trigger any fsnotify events, so I added my own event: ChangedContent which should be used very carefully and only when you really need it. But this module simplifies usage of fsnotify... Later on when we will need it it's very easy to add a separated handler for fsnotify events and using interface of this module it's very easy to subscribe to different events. I added all these constants for backward compatibility with fsnotify package.

Reply:
done